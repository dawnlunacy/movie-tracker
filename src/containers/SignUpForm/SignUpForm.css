form {
    align-items: center;
    background-color: #161616;
    border: 5px solid white;
    background-image: url('../../images/background-curtain.jpg');
    background-repeat: no-repeat;
    background-size: 100% 100%;
    display: flex;
    flex-direction: column;
    height:700px;
    padding: 30px;
    width: 800px;
    margin: auto;
    margin-top: 50px;
}

input, label {
    display: block;
    text-align: center;
    margin: 10px
}

input {
    width: 250px;
    height: 35px;
    border-radius: 5px;
}

h2 {
    font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

h3 {
    color: #CC0000;
}

button {
    margin: 10px;
}


