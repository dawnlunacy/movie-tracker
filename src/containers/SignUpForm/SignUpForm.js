import React, { Component } from 'react';
import { getUser } from '../../utils/apiCalls';
import { connect } from 'react-redux';
import { saveUser } from '../../actions/index';
import './SignUpForm.css';

class SignUpForm extends Component {
    constructor() {
        super();
        this.state = {
          newUserInput: {
            name: '',
            email: '',
            password: '',
          },
          newUser: {
            name: '',
            email: '',
            id: null
          },
          error: ''
        }
    }

    handleChange = (e) => {
      this.setState({error: ''})
      let newUserInfo = this.state.newUserInput;
      newUserInfo = {...newUserInfo, [e.target.name]: e.target.value}
        this.setState({newUserInput: newUserInfo})
    }

    submitForm = async (e) => {
        e.preventDefault();
        const createUser = await getUser(this.state.newUserInput, 'http://localhost:3001/api/v1/users')
          if (!createUser.ok) {
            const error = await createUser.json()
            console.log("ERROR in signup", error.error.detail)
            if (error.error.detail.includes('email')) {
              this.setState({error: " That email is already taken "|| ''})
            }
          } else {
            const newUser = await createUser.json()
            console.log("new User", newUser)
            this.setState({newUser:newUser || ''})
          }
        this.resetInputs()
    }

    resetInputs = () => {
      this.setState({newUserInput: {
        name: '',
        email: '',
        password: '',
      }})

    }

    render() {
        return (
            <>
            <form>
              <div className="sign-up-background">
                <h2> Sign Up </h2>
                <input
                  className="name-input"
                  type="text"
                  placeholder="Enter Name"
                  name="name"
                  value={this.state.newUserInput.name}
                  onChange={this.handleChange}
                />
                <input
                  className="email-input"
                  type="text"
                  placeholder="Enter Email"
                  name="email"
                  value={this.state.newUserInput.email}
                  onChange={this.handleChange}
                />
                <input
                  className="password-input"
                  type="text"
                  placeholder="Enter Password"
                  name="password"
                  value={this.state.newUserInput.password}
                  onChange={this.handleChange}
                />
                <button onClick={(e) => this.submitForm(e)}> SIGN UP </button>
                <p> {this.state.newUser.name} </p>
                <h3> {this.state.error} </h3>
              </div>
            </form>
            </>
        )
    }
}

export const mapDispatchToProps = dispatch => ({
  saveUser: currentUser => dispatch(saveUser(currentUser))
})

export default connect(null, mapDispatchToProps)(SignUpForm);