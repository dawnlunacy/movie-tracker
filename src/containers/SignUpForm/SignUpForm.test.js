import React from 'react';
import SignUpForm from './SignUpForm';
import { shallow } from 'enzyme';

describe('SignUpForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SignUpForm/>); 
  });

  it('should match snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('should update name in state when handleChange is called', () => {
    const mockEvent = {
      target: {
        name: 'name',
        value: 'Fernando'
      }
    }

    // wrapper.instance().handleChange = jest.fn();
    wrapper.instance().handleChange(mockEvent)

    expect(wrapper.state('name')).toEqual('Fernando')
  })

})