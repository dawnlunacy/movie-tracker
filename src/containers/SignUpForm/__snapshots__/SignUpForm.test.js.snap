// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should match snapshot 1`] = `undefined`;
